src/substrait/textplan/tests/RoundtripTest.cpp

```diff
@@ -0,0 +1,119 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#include <algorithm>
+#include <filesystem>
+#include <memory>
+#include <string>
+#include <utility>
+
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+#include <protobuf-matchers/protocol-buffer-matchers.h>
+
+#include "absl/strings/str_split.h"
+#include "gmock/gmock.h"
+#include "substrait/textplan/SymbolTablePrinter.h"
+#include "substrait/textplan/converter/LoadBinary.h"
+#include "substrait/textplan/converter/ParseBinary.h"
+#include "substrait/textplan/parser/ParseText.h"
+#include "substrait/textplan/tests/ParseResultMatchers.h"
+
+using ::protobuf_matchers::EqualsProto;
+using ::protobuf_matchers::IgnoringFieldPaths;
+using ::protobuf_matchers::Partially;
+using ::testing::AllOf;
+
+namespace io::substrait::textplan {
+namespace {
+
+std::string addLineNumbers(const std::string& text) {
+  std::stringstream result;
+  int lineNum = 0;
+  for (absl::string_view sp : absl::StrSplit(text, '\n')) {
+    result << std::setw(4) << ++lineNum << " " << sp << std::endl;
+  }
+  return result.str();
+}
+
+class RoundTripBinaryToTextFixture
+    : public ::testing::TestWithParam<std::string> {};
+
+std::vector<std::string> getTestCases() {
+  const std::filesystem::path currPath = std::filesystem::current_path();
+  std::vector<std::string> filenames{};
+  std::filesystem::path testDataPath = currPath;
+  testDataPath.append("data");
+  for (auto const& dirEntry :
+       std::filesystem::recursive_directory_iterator{testDataPath}) {
+    std::string pathName = dirEntry.path();
+    if (pathName.substr(pathName.length() - 5) == ".json") {
+      filenames.push_back(pathName);
+    }
+  }
+  std::sort(filenames.begin(), filenames.end());
+  return filenames;
+}
+
+TEST_P(RoundTripBinaryToTextFixture, RoundTrip) {
+  auto filename = GetParam();
+  std::string json = readFromFile(filename);
+  auto planOrErrors = loadFromJson(json);
+  std::vector<std::string> errors = planOrErrors.errors();
+  ASSERT_THAT(errors, ::testing::ElementsAre());
+
+  auto plan = *planOrErrors;
+
+  auto textResult = parseBinaryPlan(plan);
+  auto symbols = textResult.getSymbolTable().getSymbols();
+
+  std::string outputText =
+      SymbolTablePrinter::outputToText(textResult.getSymbolTable());
+
+  auto stream = loadTextString(outputText);
+  try {
+    auto result = parseStream(stream);
+    auto outputBinary =
+        SymbolTablePrinter::outputToBinaryPlan(result.getSymbolTable());
+
+    ASSERT_THAT(
+        result,
+        ::testing::AllOf(
+            ParsesOk(),
+            HasErrors({}),
+            AsBinaryPlan(IgnoringFieldPaths(
+                {"extension_uris", "extensions"}, EqualsProto(plan)))))
+        << std::endl
+        << "Intermediate result:" << std::endl
+        << addLineNumbers(outputText);
+  } catch (std::invalid_argument ex) {
+    FAIL() << ex.what();
+  }
```

Comment:
What is this catching?  Maybe use ASSERT_NO_THROW?

Reply:
Done.